src/corehost/cli/deps_resolver.cpp

```diff
@@ -633,11 +633,11 @@ void deps_resolver_t::resolve_additional_deps(const hostpolicy_init_t& init)
         {
             for (int i = 1; i < m_fx_definitions.size(); ++i)
             {
-                // We'll search deps files in 'base_dir'/shared/fx_name/fx_ver
+                // We'll search deps files in 'base_dir'/shared/fx_name/fx_requested_ver
                 pal::string_t additional_deps_path_fx = additional_deps_path;
                 append_path(&additional_deps_path_fx, _X("shared"));
                 append_path(&additional_deps_path_fx, m_fx_definitions[i]->get_name().c_str());
-                append_path(&additional_deps_path_fx, m_fx_definitions[i]->get_found_version().c_str());
+                append_path(&additional_deps_path_fx, m_fx_definitions[i]->get_requested_version().c_str()); // Use requested version as that is what the app deployed with
```

Comment:
How does this work with patch roll-forward?  If the app was built for `2.0.0`, but only `2.0.5` was installed on the machine, will the additional deps feature break?

Reply:
I'm thinking of the existing use-cases today where ASP.NET installs the `additionalDeps` folder.  See 
`C:\Program Files\dotnet\additionalDeps`.

The app developer doesn't create the folder structure.  The whole point of the additional deps feature is that the app developer doesn't know anything about these additional dependencies.  They are injected by the host (like VS or Azure).  So if an Azure machine only have `2.0.5` installed, and a developer deploys their web app to that machine, the additional deps feature won't work anymore.  The `runtimeconfig.json` will have `2.0.0` (because that's what the SDK injects by default, no matter the runtime used).  This means this code will look for `2.0.0` and it won't be there.

/cc @pakrym

Reply:
We would have to keep shipping additionaldeps for every patch version of shared runtime we ever produced even if we don't ship the runtime any more in case someone got rolled forward of his original requested version. Sounds pretty broken.

Reply:
I'll verify 2.0 semantics and get back. For now, I assume we want to emulate 2.0 semantics.

Reply:
Thinking more about it if the alternative is rolling to 2.1 and getting into mixed version graph state locking to requested version might not be that bad, considering that by default everyone gets `2.0` as a target runtime.

We might also want to update additionalDeps generation process to produce two versions of deps files - for `2.0` and the latest runtime.

Reply:
The behavior for 2.0.x is to use the "requested" version for the additional deps folder. i.e. 2.0.x uses the additional deps folder that exactly matches the version of the framework specified in the runtimeconfig.json, and does not use the "found" folder name of any patch (or minor) roll-forward.

@eerhardt - when we talked, you hard a different behavior. To emulate true 2.0.x behavior, try disabling (e.g. prefix with an underscore) out any `c:\Program Files\dotnet\host\fxr\2.1.*` folders as we always pick the latest fxr.

Here's my repro steps to verify:

0) Disable any 2.1* folders in `C:\Program Files\dotnet\host\fxr\` as mentioned above
1) Create the folder `c:\ReproDepsIssue\shared\Microsoft.NETCore.App\2.0.0`
2) Add a foo.deps.json file there (empty is fine)
3) Set `DOTNET_ADDITIONAL_DEPS=c:\ReproDepsIssue`
4) Set `COREHOST=TRACE=1`
5) Ensure you have 2.0.5 (the latest 2.0.x) release or a 2.0.x release other than 2.0.0 so we get a patch roll-forward
6) Run an app that targets 2.0.0 (has version=2.0.0 in its runtimeconfig.json) and redirect stderr to make it easier to search (e.g. `dotnet exec myapp.dll 2>out.txt`)
7) Search the stderr for "Using D:\git\repros\console4\bin\Debug\netcoreapp2.0\console4.deps.json deps file

You should see the text: "Using specified additional deps.json: 'c:\ReproDepsIssue\shared\Microsoft.NETCore.App\2.0.0\foo.deps.json'" meaning that it looked in the 2.0.0 folder even though a roll-forward occurred from 2.0.0 to 2.0.5

Based on this, I'm inclined to keep the "requested" semantics (as the PR is currently). However if we do that then I need to change the 2.1 hostfxr.dll to pass the "requested" version (and not "found") to the older 2.0.x hostpolicy.dll based on the issue in repro step 0 above.

Reply:
I just tested it on a machine that only ever had `2.0.3` installed on it, and it works like you describe above - the host only looks for an additionalDeps file in `2.0.0`, since that is what is in the runtimeconfig.json.  I wonder if that was intentional or not, as it doesn't seem correct in my mind.

Note - even though I only installed `2.0.3` on the machine, it appears that ASP.NET installs both `2.0.0` and `2.0.3` folders into the `additionalDeps` folder.